examples/headland_inversion/inversion_tools_vel.py

```diff
@@ -0,0 +1,827 @@
+import firedrake as fd
```

Comment:
This needs some more thought. It seems this module is just a copy of the original inversion_tools.py with modifications needed to accommodate vector values instead of scalars. This unfortunately not the way to develop code in a sustainable way, as we have now effectively doubled the amount of code that needs to be maintained. What happens if we further down the line want to further develop/fix bugs/make changes in the scalar inversion tools  - or there are interface changes in the internal bits of Thetis that are being used in this module. Do I fix both modules, or do we let things diverge? I also can't easily see where this module is actually different from the original. As a general rule, "Copying code" is almost never the answer - except maybe sometimes for quickly getting something to work, but then if you want to keep it, you need to go back and figure out how to refactor and integrate the two *without duplication*

Reply:
Yes, agree with everything you've said. It is indeed just a copy with modifications for outputting the control and gradients as well as updating from elevation -> velocity.

I intentionally didn't merge this script with the base one because I wasn't sure how would be best to go about certain aspects:

- Do we assume that the calibration parameter is always elevation or velocity (probably) and then automatically detect whether we are dealing with a scalar or vector field? What if we want to use both elevation and velocity data for calibration? Do we limit it to one just now and leave this as a future problem?
-  Is the way I am exporting things optimal? It's a bit of a pain because exporting is only set up for when the control parameter is the whole field, but the way I have dealt with it feels horrible. It seems like exporting might be nicer outside of the InversionManager and StationObservationManager classes... 

I will work on this on the second batch of commits!

Reply:
Made the final updates to deal with this. It is assumed that the field for loss evaluation is **either** velocity or elevation (default). The exporting is fine in any case, though I'd still be tempted to update it in the future.

One extra thing I did was change the base `DiagnosticCallback` so that the simulation start time is an attribute when exporting to hdf5. This is easier/more direct to load than taking the units string of the variable data and something that a few people had asked me about.